ships/0024-surfacing-errors-to-buildrun.md

```diff
@@ -0,0 +1,186 @@
+<!--
+Copyright The Shipwright Contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+---
+title: surfacing-errors-to-buildrun
+authors:
+  - "@dalbar"
+reviewers:
+  - "@qu1queee"
+  - "@HeavyWombat"
+approvers:
+  - "@adambkaplan"
+  - "@sbose78"
+creation-date: 2021-09-10
+last-updated: 2021-09-10
+status: draft
+
+---
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Summary
+
+This ship provides a mechanism to surface errors in the build process
+into BuildRun objects granting more visibility into
+without being Tekton/Implementation specific. To
+achieve this goal, we introduce two new generic results
+`shp-error-reason` and `shp-error-message`, that a TaskRun can emit
+and a BuildRun can pick up. 
```

Comment:
I think that these should be more generic, which would allow for surfacing not only error messages and reasons, but success messages and reasons (and anything in between).